core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/WithClauseContainer.kt

```diff
@@ -21,17 +24,18 @@ internal abstract class WithClauseContainer(
   }
 
   protected fun SqlWithClause.tablesExposed(): List<LazyQuery> {
-    return cteTableNameList.zip(compoundSelectStmtList)
-        .map { (name, selectStmt) ->
-          LazyQuery(name.tableName) {
-            val query = QueryResult(name.tableName,
-                selectStmt.queryExposed().flatMap { it.columns })
-            return@LazyQuery if (name.columnAliasList.isNotEmpty()) {
-              QueryResult(name.tableName, name.columnAliasList.asColumns())
-            } else {
-              query
-            }
-          }
+    return cteTableNameList.zip(withClauseAuxiliaryStmtList)
+      .mapNotNull { (name, withClauseAuxiliaryStmt) ->
+        PsiTreeUtil.findChildOfAnyType(withClauseAuxiliaryStmt, SqlCompoundSelectStmt::class.java, PostgreSqlReturningClause::class.java)
```

Comment:
same here for `filter { is QueryElement }`

Reply:
Fixed in https://github.com/AlecStrong/sql-psi/pull/170/commits/0647c327e733e690fba8344a4bc4c244e24f7dc9.